contributor-book/src/guides/adding-a-new-operation-to-burn.md

```diff
@@ -0,0 +1,216 @@
+are for, involves a bit more indirection. Fusion, like autodiff, is not a target backend as much as
+a backend that enables certain functionality for other backends, in this case kernel fusion (which
+is currently only supported for `burn-wgpu`), so adding the operator won't involve doing any
+calculation, you'll just be describing how the generated code should look. Most of this can be
+copy/pasted/adjusted from other functions.
+
+here's how powf was added to burn fusion:
+
+1. added powf to the float ops under
+   [`burn-fusion/src/ops/float.rs`](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/ops/float.rs#L1758)
+2. added powf to the `FloatOperationDescription` enum under
+   [burn-fusion/src/stream/operation.rs](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/stream/operation.rs#L385)
+3. added powf to the implementations of `FloatOperationDescription` enum under
+   [burn-fusion/src/stream/context.rs](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-fusion/src/stream/context.rs#L726)
+
+Adding pow to wgpu was actually pretty easy due to the design. Element-wise tensor ops are just
+vectorized scalar ops, and given that raising a tensor to a scalar power prior to the tensor
+version, I just reused the code for scalar powf.
+
+here is where code was added
+
+1. to the implementation of
+   [`TensorOps` under `burn-wgpu/src/ops/float_ops.rs`](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-wgpu/src/ops/float_ops.rs#L513)
+2. the function being called was added to
+   [burn-wgpu/src/ops/numeric.rs](https://github.com/tracel-ai/burn/blob/0368409eb3a7beaeda598c0c8ce1dc0c2c8c07cc/burn-wgpu/src/ops/numeric.rs#L199)
+3. the call to the fmt function use to generate wgsl code in
+   [`burn-wgpu/src/codegen/kernel.rs`](https://github.com/tracel-ai/burn/blob/main/burn-wgpu/src/codegen/kernel.rs#L208)
+4. A custom function generator was added to
+   [`burn-wgpu/src/codegen/function.rs`](https://github.com/tracel-ai/burn/blob/main/burn-wgpu/src/codegen/function.rs#L99)
+
+Much of the logic for powf had already been defined, so not much needed to be added. The reason for
+the custom powf function is generated, rather than directly using the underlying function as is the
+case with other operations, is due to issues with case handling of the wgsl pow function, like 0 to
+the 0 power being 1, and any negative number to an even power being positive. We reused as much as
+the existing logic as possible, such as the operation output description generation in
+[`burn-wgpu/src/fusion/elemwise/builder.rs`](https://github.com/tracel-ai/burn/blob/main/burn-wgpu/src/fusion/elemwise/optimization.rs)
+and then branched at the last point based off the var type of the rhs. I don't know if there are
+other places within `burn-wgpu` where additions would have been necessary otherwise.
```

Comment:
Not sure the last sentence is necessary.

Reply:
fixed in next push